tmt/steps/execute/scripts/tmt-report-result

```diff
@@ -11,6 +11,61 @@ help=False
 die() { echo "$*" >&2; exit 2; }  # complain to STDERR and exit with error
 needs_arg() { if [ -z "$OPTARG" ]; then die "No arg for --$OPT option"; fi; }
 
+write_report_file () {
+    rm -f $REPORT_RESULT_OUTPUTFILE
+    echo "SERVER=$server" >> $REPORT_RESULT_OUTPUTFILE
+    echo "PORT=$port" >> $REPORT_RESULT_OUTPUTFILE
+    echo "MESSAGE=$message" >> $REPORT_RESULT_OUTPUTFILE
+    echo "OUTPUTFILE=$outputFile" >> $REPORT_RESULT_OUTPUTFILE
+    echo "DISABLEPLUGIN=$disablePlugin" >> $REPORT_RESULT_OUTPUTFILE
+    echo "TESTNAME=$TESTNAME" >> $REPORT_RESULT_OUTPUTFILE
+    echo "TESTRESULT=$TESTRESULT" >> $REPORT_RESULT_OUTPUTFILE
+    echo "METRIC=$METRIC" >> $REPORT_RESULT_OUTPUTFILE
+}
+
+check_report_file () {
+    declare -A RESULTS_HIERARCHY=( ["FAIL"]=4 ["WARN"]=3 ["PASS"]=2 ["SKIP"]=1 )
+    # Create the output file in which to store the results.
+    REPORT_RESULT_OUTPUTFILE="$TMT_TEST_DATA/restraint-result"
+    if [ -e "$REPORT_RESULT_OUTPUTFILE" ]; then
+        # Check the existing output file.
+        # If new result is higher on the hierarchy
+        # or no result is contained in the file
+        # then overwrite the file with the new
+        # report contents.
+        fileResult=$(/usr/bin/grep "TESTRESULT" $REPORT_RESULT_OUTPUTFILE |
+        cut -d = -f 2)
+        if [ ! $fileResult ]; then
+            write_report_file;
+            exit 0
+        elif [ ! $TESTRESULT ]; then
+            exit 0
+        fi
+        fileHierarchy="${RESULTS_HIERARCHY[$fileResult]}"
+        # Compare the current result hierarchy against
+        # the hierarchy result retrieved from the file.
+        # Overwrite the file if the current hierachy is higher.
+        thisResultHierarchy="${RESULTS_HIERARCHY[$TESTRESULT]}"
+        if [ $thisResultHierarchy -gt $fileHierarchy ]; then
+            write_report_file
+            exit 0
+        fi
+        exit 0
+    else
+        # Create the output file with the report contents.
+        write_report_file
+        exit 0
+    fi
+}
+
+if [[ "$0" == *"rhts-report-result"* ]]; then
```

Comment:
Does this match in some configuration? I've tried provided reproducer but `$0` had the value `/usr/local/bin/rhts-report-result` during the execution.

Reply:
The '*' wildcard allows the matching of all characters before the 'rhts-report-result' string, and as such it matches with the command regardless of the origin directory.

> [root@fedora scripts]# /usr/local/bin/rhts-report-result /smoke_test PASS rhts_report 69
> [root@fedora scripts]# cat /tmp/restraint-result 
> SERVER=
> PORT=
> MESSAGE=
> OUTPUTFILE=rhts_report
> DISABLEPLUGIN=
> TESTNAME=/smoke_test
> TESTRESULT=PASS
> METRIC=69
> [root@fedora scripts]#

> [root@fedora scripts]# rhts-report-result /second_smoke_test PASS rhts_report 75
> [root@fedora scripts]# cat /tmp/restraint-result 
> SERVER=
> PORT=
> MESSAGE=
> OUTPUTFILE=rhts_report
> DISABLEPLUGIN=
> TESTNAME=/second_smoke_test
> TESTRESULT=PASS
> METRIC=75
> [root@fedora scripts]#